modules/cugraph/src/graph_coo.cpp

```diff
@@ -56,13 +67,10 @@ GraphCOO::GraphCOO(CallbackArgs const& args) : EnvLocalObjectWrap<GraphCOO>(args
 
 size_t GraphCOO::num_nodes() {
   if (!node_count_computed_) {
-    auto const& src      = *src_.Value();
-    auto const& dst      = *dst_.Value();
-    auto const src_max   = src.minmax().second;
-    auto const dst_max   = dst.minmax().second;
-    node_count_          = 1 + std::max<int32_t>(  //
-                        src_max->get_value().ToNumber(),
-                        dst_max->get_value().ToNumber());
+    auto max_id = [&](Napi::Reference<Column::wrapper_t> const& col) -> int32_t {
+      return col.Value()->minmax().second->get_value().ToNumber();
+    };
```

Comment:
Wasn't there some obscure problem related to lifetimes that precipitated the old step-by-step code structure? Does this new version with the lambda avoid that?

Reply:
No there isn't, but I refactored it into a lambda to verify that, and left it since it's a bit cleaner.